phoenix-core/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -168,7 +169,11 @@ public QueryPlan compileUnionAll(SelectStatement select) throws SQLException {
             SelectStatement subSelect = unionAllSelects.get(i);
             // Push down order-by and limit into sub-selects.
             if (!select.getOrderBy().isEmpty() || select.getLimit() != null) {
```

Comment:
Does this if statement need to be change to this (and if so, check for other occurrences)?

    if (!select.getOrderBy().isEmpty() || select.getLimit() != null || select.getOffset() != null)